test/test_complate.js

```diff
@@ -7,17 +7,32 @@ suite("complate extension");
 
 test("rendering", async () => {
 	let bundle = new Bundle(fixturePath());
-	let html = await bundle.renderString(`
+	let jsx = `
 import Article from "./article.jsx";
 
-<>
+<div>
 	<h1>Hello World</h1>
 
-	<Article title="Lipsum" tags={["foo", "bar"]}>
-		<p>lorem ipsum dolor sit amet</p>
-	</Article>
-</>
-	`, "snippet.jsx");
-	assertSame(html.replace(/\n|\t/g, ""), // eslint-disable-next-line max-len
-			"<h1>Hello World</h1><article><h2>Lipsum</h2><ol><li>foo</li><li>bar</li></ol><p>lorem ipsum dolor sit amet</p></article>");
+	<>
+		<Article title="Lipsum" tags={["foo", "bar"]}>
+			<p>lorem ipsum dolor sit amet</p>
+		</Article>
+	</>
+</div>
+  `;
+	let html = await bundle.renderString(jsx, "snippet.jsx");
+	let expected = `<div>
+	<h1>Hello World</h1>
+
+	
+		<article>
+		<h2>Lipsum</h2>
+		<ol><li>foo</li><li>bar</li></ol>
+		
+			<p>lorem ipsum dolor sit amet</p>
+		
+	</article>
+	
+</div>`;
```

Comment:
all that whitespace is a bit disturbing - not just in terms of FND aesthetics, but also because it might lead to undesirable results, particularly because JSX usually has the opposite problem: https://www.bennadel.com/blog/2880-a-quick-look-at-rendering-white-space-using-jsx-in-reactjs.htm

Reply:
What do you suggest as an alternative? I'm cautious about removing whitespace, e.g. because <pre> etc. need it.

Reply:
As discussed (dug up by yourself): https://reactjs.org/docs/jsx-in-depth.html#string-literals-1 should suffice as a reference, perhaps also https://github.com/babel/babel/blob/e7961a08a839b0bfe2c5a08f2e1c7e3d436af144/packages/babel-types/src/utils/react/cleanJSXElementLiteralChild.js.

Reply:
larsrh/complate-ast#25